### ISSUE TITLE
feat: IntegerMatrix should supports mpz

### ISSUE THREAD

**Issue by alxiong** [fplll/fpylll#268]
I was reading [`assign_mpz`](https://github.com/fplll/fpylll/blob/master/src/fpylll/io.pyx#L40-L60) and see that even if internally the IntegerMatrix uses mpz, it doesn't accept value passed in as type `mpz`.

ideally I want to do:

```python
from fpylll import IntegerMatrix
from gmpy2 import mpz

A = IntegerMatrix(5, 5)
A.gen_identity(5)
a = mpz(1000)
A[2, 3] = a
```

**Comment by alxiong** [comment]
this should do the work:

https://github.com/alxiong/fpylll/commit/7f994c1c0deda2b3e0f712c641da9a12af6d98d2

tested locally, i can compile `fpylll` and also pass in `gmpy2.mpz` values.

**Comment by cr-marcstevens** [comment]
I kind of agree with @malb here: are we going to support every multiprec python option by bringing it in as a requirement for fpylll?

My suggestion would be to look at a generic solution that works for any bigint type of python object by manually chopping it in limbs and reconstruct it in the fplll mpz.

**Comment by alxiong** [comment]
i agree with your concern, and don't have any strong opinion on how should fpylll handles these. 

feel free to close this issue. 

(i was only hoping to use this issue as future reference in case anyone wants to use `gmpy2` library, since sagemath couldn't be easily installed as PyPI package like `gmpy2` does. 
at least the 3 lines of code above, showcase how one can modify the `fpylll` to support that)

**Comment by malb** [comment]
I wonder if we could add Marc's idea though: an interface that takes an input and repeatedly considers it mod MAX_INT_SIZE to construct the internal integer? This then may or may not be faster than just reading in a string?
